File: Python/Find Minimum in Rotated Sorted Array.py
class Solution:
    # @param num, a list of integer
    # @return an integer
    def findMin(self, num):
        tag = -1
        for i in range(len(num) - 1):
            if num[i] > num[i+1]:
                tag = i + 1
                break
        if tag != -1:
            return num[tag]
        else:
            return num[0]

File: C++/Find Minimum in Rotated Sorted Array.cpp
#include <vector>

class Solution {
public:
    int findMin(vector<int> &num) {
        int len = num.size();
        int left = 0, right = len - 1, mid;
        while(left < right && num[left] > num[right]){
            mid = (left + right) / 2;
            if(num[mid] < num[right]) right = mid;
            else left = mid + 1;
        }
        return num[left];
    }
};